httpclient-backend/monix/src/main/scala/sttp/client3/httpclient/monix/HttpClientMonixBackend.scala

```diff
@@ -49,47 +50,32 @@ class HttpClientMonixBackend private (
         monad.eval(BodyPublishers.fromPublisher(FlowAdapters.toFlowPublisher(stream.toReactivePublisher)))
     }
 
-  override protected val bodyFromHttpClient: BodyFromHttpClient[Task, MonixStreams, InputStream] =
-    new InputStreamBodyFromHttpClient[Task, MonixStreams] {
-      override def inputStreamToStream(is: InputStream): Task[(streams.BinaryStream, () => Task[Unit])] = {
-        Task.eval {
-          (
-            Observable
-              .fromInputStream(Task.now(is))
-              .map(ByteBuffer.wrap)
-              .guaranteeCase(_ => Task(is.close())),
-            () => Task.eval(is.close())
-          )
-        }
-      }
-      override val streams: MonixStreams = MonixStreams
-      override implicit def monad: MonadError[Task] = TaskMonadAsyncError
-      override def compileWebSocketPipe(
-          ws: WebSocket[Task],
-          pipe: Observable[WebSocketFrame.Data[_]] => Observable[WebSocketFrame]
-      ): Task[Unit] = MonixWebSockets.compilePipe(ws, pipe)
+  override protected val bodyFromHttpClient: BodyFromHttpClient[Task, MonixStreams, MonixStreams.BinaryStream] =
+    new MonixBodyFromHttpClient {
+      override implicit def scheduler: Scheduler = s
+      override implicit def monad: MonadError[Task] = responseMonad
     }
 
   override protected def createSimpleQueue[T]: Task[SimpleQueue[Task, T]] =
     Task.eval(new MonixSimpleQueue[T](None))
 
-  override protected def publisherToBody(p: Publisher[util.List[ByteBuffer]]): InputStream = {
-    val subscriber = new InputStreamSubscriber
-    p.subscribe(subscriber)
-    subscriber.inputStream
+  override protected def publisherToBody(p: Publisher[util.List[ByteBuffer]]): Observable[ByteBuffer] = {
+    Observable
+      .fromReactivePublisher(p)
+      .map(list => ByteBuffer.wrap(list.asScala.toList.flatMap(_.safeRead()).toArray))
```

Comment:
this can probably be optimized, so that e.g. for a single-element list we don't convert it to array and back to a buffer.

Also, for multiple buffers, we could probably create an observable which emits all of these buffers, instead of emitting one combined array